### PR TITLE
chore: cleanup from splitting plugin-templates

### PR DESCRIPTION
**Pull request by randi274**
### What does this PR do?
Updates the repo to bring it up to standard with other salesforce CLI Plugins post-migration. Updates include:
- Add gitconfig
- Add bug/feature request/PR templates
- Add "Run Current Test", fix "Run All Tests"
- Fix debugging doc
- Check that tests pass locally
- Run basic test linked with sfdx plugins

### What issues does this PR fix or reference?
@W-11309541@

### CODE REVIEW

File: .eslintrc.json
Comment (gbockus-sf) [inline .eslintrc.json:54]: nit: should this be 2022

File: .github/CODEOWNERS
Comment (gbockus-sf) [inline .github/CODEOWNERS:8]: IS this formatted correctly?  It looks different then https://github.com/forcedotcom/salesforcedx-templates/blob/main/.github/CODEOWNERS

**Review by gbockus-sf — APPROVED** [review]
Looks good.  A couple of very small nitty things to consider.  Pulled local and rand build/test/lint and related.  Looks good to go.

File: .github/CODEOWNERS
Comment (iowillhoit) [reply .github/CODEOWNERS:8]: I think it would probably still work. Github is showing it as valid, but lets updated it to match [what is in our plugin template](https://github.com/salesforcecli/plugin-template-sf/blob/main/.github/CODEOWNERS)

**Comment by iowillhoit** [comment]
Hey @randi274 👋  Could you do a search for `salesforcedx` and remove those references. One important one is the `repository` value in the `package.json`

File: .eslintrc.json
Comment (iowillhoit) [inline .eslintrc.json:1]: Most places we have removed the `.eslintrc.json` file and replaced it with a `.eslintrc.js` file like [this one](https://github.com/salesforcecli/plugin-template-sf/blob/main/.eslintrc.js). You can still add custom rules to that if needed ([example](https://github.com/salesforcecli/plugin-release-management/blob/580aabf621deed609ebc9afbd5cbc0ef191a40ba/.eslintrc.js#L9))

File: package.json
Comment (iowillhoit) [inline package.json:4]: Update to `lib/index.js`

File: package.json
Comment (iowillhoit) [inline package.json:3]: Add a top-level `description` please 🙏

File: package.json
Comment (iowillhoit) [inline package.json:9]: Update to `tslib ^2` please (Github will not let me select that line for a comment 🤔)

File: package.json
Comment (iowillhoit) [inline package.json:143]: We've removed our husky config from the `package.json` and moved them to a `.husky` directory. [Example here](https://github.com/salesforcecli/plugin-template-sf/tree/main/.husky)

**Comment by iowillhoit** [comment]
You'll need to add all of the plugin's `topics` and `subtopics` in the `package.json` for the `--help` to work correctly. If you run `bin/dev force --help` the only (sub)topic that shows up is `force:analytics`. [Here is an example](https://github.com/salesforcecli/plugin-data/blob/8d92caf08e9bcf67955730b7ba19335eb5306adb/package.json#L29) of some nested topics. 

FWIW, you don't need to worry about adding a `longDescription`, just `description` is fine.

File: tsconfig.settings.json
Comment (iowillhoit) [inline tsconfig.settings.json:28]: Is there a reason all of this ts config is in its own file and not part of `tsconfig.json`? ([tsconfig.settings.json](https://github.com/salesforcecli/plugin-templates/pull/1/files#diff-6d9dfe806b0aeffa01011c28e580110501b307dc617ec8f214dc1c6986dbcdf4))

File: .eslintrc.json
Comment (randi274) [reply .eslintrc.json:54]: for like 20 more days!

File: .eslintrc.json
Comment (randi274) [reply .eslintrc.json:1]: Will do! I'd like to have you re-review the state of the .js file once I push the update, to make sure it all looks valid.

File: tsconfig.settings.json
Comment (randi274) [reply tsconfig.settings.json:28]: Historically, there probably was, but that knowledge is long lost now 😄 I'll try moving it and see if it's as simple as that!

File: package.json
Comment (randi274) [inline package.json:87]: @iowillhoit - let me know if this work for topics/subtopics! They were manually generated, so I'm sure are prone to user-error here.

File: package.json
Comment (randi274) [inline package.json:135]: Project + staticresource below don't have any nested folders/data types

File: package.json
Comment (randi274) [inline package.json:171]: There wasn't an option in the dev-scripts repo for this, so just added here.

File: src/utils/messageUtil.ts
Comment (randi274) [inline src/utils/messageUtil.ts:15]: Changes here on down were all lint fixes - I updated types where I could, and removed linting where I couldn't. Other errors were for improper ordering, missing return types (those ones caused actual errors when fixed), and mostly missing types resulting in rogue `any`s.

**Comment by randi274** [comment]
Hey @iowillhoit 👋  Finally circling back to this one! Is there anything else you'd like me to add before merging this in?

**Comment by iowillhoit** [comment]
Hey @randi274 👋 Could you please do a final round of QA on this? I will do the same and then I think we'll be good to merge!

**Comment by randi274** [comment]
Alright @iowillhoit things are looking groovy - did a follow-up sanity check and confirmed that we can:

- Locally create apex and LWC using default templates
- Link the plugin and create apex using custom templates

All good from my end!

**Comment by iowillhoit** [comment]
Thanks Randi! Things are looking great on my end too. Approved and ready for merge.

QA Notes:
### force:analytics:template:create
🟢 Able to create analytic templates (and set output dir)
🟢 JSON output matches

### force:apex:create
🟢 Was able to create an Apex class with every available template (and set output dir)
🟢 JSON output matches

### force:lightning
#### force:lightning:app:create
🟢 Able to create a lighting app
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
#### force:lightning:component:create
🟢 Able to create a lighting component
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
#### force:lightning:event:create
🟢 Able to create a lighting event
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
#### force:lightning:interface:create
🟢 Able to create a lighting interface
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
#### force:lightning:test:create
🟢 Able to create a lighting test
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files

### force:project:create
🟢 Able to create a project (with all templates)
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
🟢 force overwrites files changed
🟢 Allows setting `namespace`
🟢 Allows setting `defaultPackageDir`
🟢 Supports creating a manifest

### force:staticresource:create
🟢 Able to create a project (with all templates)
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
🟢 Supports setting `contentType`

### force:visualforce
#### force:visualforce:component 
🟢 Able to create a visualforce component
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
🟢 Supports setting `label`

#### force visualforce page
🟢 Able to create a visualforce component
🟢 Able to set `outputDir`
🟢 JSON output matches
🟢 Detects identical files
🟢 Supports setting `label`

**Review by iowillhoit — APPROVED** [review]

**Pull request merged**
